scheduler/util.go

```diff
@@ -700,6 +701,11 @@ func inplaceUpdate(ctx Context, eval *structs.Evaluation, job *structs.Job,
 			continue
 		}
 
+		// The alloc is on a node that's now in an ineligible DC
+		if !helper.SliceStringContains(job.Datacenters, node.Datacenter) {
+			continue
+		}
```

Comment:
As I noted in https://github.com/hashicorp/nomad/issues/10746#issuecomment-875644525 I'm not wild about the placement of this check, because it means we can't really test the behavior except in the whole-scheduler tests as I've done in `TestServiceSched_JobModify_Datacenters`